### ISSUE TITLE
crun-0:1.18.1-1.fc41.x86_64 causes issues with rootles --device integration

### ISSUE THREAD

**Issue by benjamin-salchow** [containers/podman#24517]
### Issue Description

Since I upgraded to crun-0:1.18.1-1.fc41.x86_64 I'm not able to use any /dev/ttyUSB* device in my not privileged container.

I use Fedora and have several containers that are not working any more. It's possible to reproduce it with any container image. Downgrade resolves the issue to version 1.17-1.fc41.

### Steps to reproduce the issue

Running the following container:

```bash
/usr/bin/podman run \
  --cgroups=enabled \
  --rm \
  --sdnotify=conmon \
  --replace \
  -it \
  --name test \
  --net=smarthomeNetwork \
  --device=/dev/ttyUSB0:/dev/ttyUSB0:rw \
  --device=/dev/ttyUSB1:/dev/ttyUSB1:rw \
  --group-add keep-groups \
  --annotation run.oci.keep_original_groups=1 \
  fedora:latest
```

with version: crun-0:1.18.1-1.fc41.x86_64

causes the following issue:

```bash
[root@e49a05aefa7b /]# cat /dev/ttyUSB0
cat: /dev/ttyUSB0: Operation not permitted
```

running the exact same container with version: crun-0:1.17-1.fc41.x86_64

Everything works as expected and I can use the device.

### Describe the results you received

```bash
[root@e49a05aefa7b /]# cat /dev/ttyUSB0
cat: /dev/ttyUSB0: Operation not permitted
```
It should work, because with the version before, I'm able to use all the --devices /dev/ttyUSB*

### Describe the results you expected

It should work as in version crun-0:1.17-1.fc41.x86_64

The device mounted into the container should work.

### podman info output

```yaml
Downgraded version that works: crun-0:1.17-1.fc41.x86_64

host:
  arch: amd64
  buildahVersion: 1.37.5
  cgroupControllers:
  - cpuset
  - cpu
  - io
  - memory
  - hugetlb
  - pids
  - rdma
  - misc
  cgroupManager: systemd
  cgroupVersion: v2
  conmon:
    package: conmon-2.1.12-3.fc41.x86_64
    path: /usr/bin/conmon
    version: 'conmon version 2.1.12, commit: '
  cpuUtilization:
    idlePercent: 66.04
    systemPercent: 7.83
    userPercent: 26.13
  cpus: 4
  databaseBackend: boltdb
  distribution:
    distribution: fedora
    variant: server
    version: "41"
  eventLogger: journald
  freeLocks: 1804
  hostname: homeserver
  idMappings:
    gidmap: null
    uidmap: null
  kernel: 6.11.6-300.fc41.x86_64
  linkmode: dynamic
  logDriver: journald
  memFree: 23786414080
  memTotal: 33015615488
  networkBackend: netavark
  networkBackendInfo:
    backend: netavark
    dns:
      package: aardvark-dns-1.13.0-1.fc41.x86_64
      path: /usr/libexec/podman/aardvark-dns
      version: aardvark-dns 1.13.0
    package: netavark-1.13.0-1.fc41.x86_64
    path: /usr/libexec/podman/netavark
    version: netavark 1.13.0
  ociRuntime:
    name: crun
    package: crun-1.17-1.fc41.x86_64
    path: /usr/bin/crun
    version: |-
      crun version 1.17
      commit: 000fa0d4eeed8938301f3bcf8206405315bc1017
      rundir: /run/crun
      spec: 1.0.0
      +SYSTEMD +SELINUX +APPARMOR +CAP +SECCOMP +EBPF +CRIU +LIBKRUN +WASM:wasmedge +YAJL
  os: linux
  pasta:
    executable: /usr/bin/pasta
    package: passt-0^20241030.gee7d0b6-1.fc41.x86_64
    version: |
      pasta 0^20241030.gee7d0b6-1.fc41.x86_64-pasta
      Copyright Red Hat
      GNU General Public License, version 2 or later
        <https://www.gnu.org/licenses/old-licenses/gpl-2.0.html>
      This is free software: you are free to change and redistribute it.
      There is NO WARRANTY, to the extent permitted by law.
  remoteSocket:
    exists: true
    path: /run/podman/podman.sock
  rootlessNetworkCmd: pasta
  security:
    apparmorEnabled: false
    capabilities: CAP_CHOWN,CAP_DAC_OVERRIDE,CAP_FOWNER,CAP_FSETID,CAP_KILL,CAP_NET_BIND_SERVICE,CAP_SETFCAP,CAP_SETGID,CAP_SETPCAP,CAP_SETUID,CAP_SYS_CHROOT
    rootless: false
    seccompEnabled: true
    seccompProfilePath: /usr/share/containers/seccomp.json
    selinuxEnabled: true
  serviceIsRemote: false
  slirp4netns:
    executable: ""
    package: ""
    version: ""
  swapFree: 25769795584
  swapTotal: 25769795584
  uptime: 0h 11m 40.00s
  variant: ""
plugins:
  authorization: null
  log:
  - k8s-file
  - none
  - passthrough
  - journald
  network:
  - bridge
  - macvlan
  - ipvlan
  volume:
  - local
registries:
  search:
  - registry.fedoraproject.org
  - registry.access.redhat.com
  - docker.io
store:
  configFile: /usr/share/containers/storage.conf
  containerStore:
    number: 14
    paused: 0
    running: 10
    stopped: 4
  graphDriverName: overlay
  graphOptions:
    overlay.imagestore: /usr/lib/containers/storage
    overlay.mountopt: nodev,metacopy=on
  graphRoot: /var/lib/containers/storage
  graphRootAllocated: 481408286720
  graphRootUsed: 342244376576
  graphStatus:
    Backing Filesystem: xfs
    Native Overlay Diff: "false"
    Supports d_type: "true"
    Supports shifting: "true"
    Supports volatile: "true"
    Using metacopy: "true"
  imageCopyTmpDir: /var/tmp
  imageStore:
    number: 48
  runRoot: /run/containers/storage
  transientStore: false
  volumePath: /var/lib/containers/storage/volumes
version:
  APIVersion: 5.2.5
  Built: 1729209600
  BuiltTime: Fri Oct 18 02:00:00 2024
  GitCommit: ""
  GoVersion: go1.23.2
  Os: linux
  OsArch: linux/amd64
  Version: 5.2.5


After dnf upgrade to:

Package                                                                          Arch              Version                                                                          Repository                                          Size
Upgrading:
 crun                                                                            x86_64            1.18.1-1.fc41                                                                    updates                                        532.0 KiB
   replacing crun                                                                x86_64            1.17-1.fc41                                                                      fedora                                         519.7 KiB




It's not working and output of podman info is:

host:
  arch: amd64
  buildahVersion: 1.37.5
  cgroupControllers:
  - cpuset
  - cpu
  - io
  - memory
  - hugetlb
  - pids
  - rdma
  - misc
  cgroupManager: systemd
  cgroupVersion: v2
  conmon:
    package: conmon-2.1.12-3.fc41.x86_64
    path: /usr/bin/conmon
    version: 'conmon version 2.1.12, commit: '
  cpuUtilization:
    idlePercent: 22.45
    systemPercent: 21.55
    userPercent: 56
  cpus: 4
  databaseBackend: boltdb
  distribution:
    distribution: fedora
    variant: server
    version: "41"
  eventLogger: journald
  freeLocks: 1803
  hostname: homeserver
  idMappings:
    gidmap: null
    uidmap: null
  kernel: 6.11.6-300.fc41.x86_64
  linkmode: dynamic
  logDriver: journald
  memFree: 29886099456
  memTotal: 33015599104
  networkBackend: netavark
  networkBackendInfo:
    backend: netavark
    dns:
      package: aardvark-dns-1.13.0-1.fc41.x86_64
      path: /usr/libexec/podman/aardvark-dns
      version: aardvark-dns 1.13.0
    package: netavark-1.13.0-1.fc41.x86_64
    path: /usr/libexec/podman/netavark
    version: netavark 1.13.0
  ociRuntime:
    name: crun
    package: crun-1.18.1-1.fc41.x86_64
    path: /usr/bin/crun
    version: |-
      crun version 1.18.1
      commit: c41f034fdbb9742c395085fc98459c94ad1f9aae
      rundir: /run/crun
      spec: 1.0.0
      +SYSTEMD +SELINUX +APPARMOR +CAP +SECCOMP +EBPF +CRIU +LIBKRUN +WASM:wasmedge +YAJL
  os: linux
  pasta:
    executable: /usr/bin/pasta
    package: passt-0^20241030.gee7d0b6-1.fc41.x86_64
    version: |
      pasta 0^20241030.gee7d0b6-1.fc41.x86_64-pasta
      Copyright Red Hat
      GNU General Public License, version 2 or later
        <https://www.gnu.org/licenses/old-licenses/gpl-2.0.html>
      This is free software: you are free to change and redistribute it.
      There is NO WARRANTY, to the extent permitted by law.
  remoteSocket:
    exists: true
    path: /run/podman/podman.sock
  rootlessNetworkCmd: pasta
  security:
    apparmorEnabled: false
    capabilities: CAP_CHOWN,CAP_DAC_OVERRIDE,CAP_FOWNER,CAP_FSETID,CAP_KILL,CAP_NET_BIND_SERVICE,CAP_SETFCAP,CAP_SETGID,CAP_SETPCAP,CAP_SETUID,CAP_SYS_CHROOT
    rootless: false
    seccompEnabled: true
    seccompProfilePath: /usr/share/containers/seccomp.json
    selinuxEnabled: true
  serviceIsRemote: false
  slirp4netns:
    executable: ""
    package: ""
    version: ""
  swapFree: 25769795584
  swapTotal: 25769795584
  uptime: 0h 1m 49.00s
  variant: ""
plugins:
  authorization: null
  log:
  - k8s-file
  - none
  - passthrough
  - journald
  network:
  - bridge
  - macvlan
  - ipvlan
  volume:
  - local
registries:
  search:
  - registry.fedoraproject.org
  - registry.access.redhat.com
  - docker.io
store:
  configFile: /usr/share/containers/storage.conf
  containerStore:
    number: 13
    paused: 0
    running: 9
    stopped: 4
  graphDriverName: overlay
  graphOptions:
    overlay.imagestore: /usr/lib/containers/storage
    overlay.mountopt: nodev,metacopy=on
  graphRoot: /var/lib/containers/storage
  graphRootAllocated: 481408286720
  graphRootUsed: 342251941888
  graphStatus:
    Backing Filesystem: xfs
    Native Overlay Diff: "false"
    Supports d_type: "true"
    Supports shifting: "true"
    Supports volatile: "true"
    Using metacopy: "true"
  imageCopyTmpDir: /var/tmp
  imageStore:
    number: 48
  runRoot: /run/containers/storage
  transientStore: false
  volumePath: /var/lib/containers/storage/volumes
version:
  APIVersion: 5.2.5
  Built: 1729209600
  BuiltTime: Fri Oct 18 02:00:00 2024
  GitCommit: ""
  GoVersion: go1.23.2
  Os: linux
  OsArch: linux/amd64
  Version: 5.2.5
```

### Podman in a container

No

### Privileged Or Rootless

Rootless

### Upstream Latest Release

Yes

### Additional environment details

Fedora Server 41

### Additional information

Workaround ATM for me is to:

```bash
dnf downgrade crun
```

output:
```bash
Updating and loading repositories:
Repositories loaded.
Package                                                                          Arch              Version                                                                          Repository                                          Size
Downgrading:
 crun                                                                            x86_64            1.17-1.fc41                                                                      fedora                                         519.7 KiB
   replacing crun                                                                x86_64            1.18.1-1.fc41                                                                    updates                                        532.0 KiB

Transaction Summary:
 Replacing:          1 package
 Downgrading:        1 package

Total size of inbound packages is 229 KiB. Need to download 229 KiB.
Is this ok [y/N]: y
[1/1] crun-0:1.17-1.fc41.x86_64                                             100% | 941.4 KiB/s | 228.8 KiB |  00m00s
--------------------------------------------------------------------------------------------------------------------
[1/1] Total                                                                 100% | 665.0 KiB/s | 228.8 KiB |  00m00s
Running transaction
[1/4] Verify package files                                                  100% | 125.0   B/s |   1.0   B |  00m00s
[2/4] Prepare transaction                                                   100% |   4.0   B/s |   2.0   B |  00m00s
[3/4] Downgrading crun-0:1.17-1.fc41.x86_64                                 100% |   7.9 MiB/s | 520.8 KiB |  00m00s
[4/4] Erasing crun-0:1.18.1-1.fc41.x86_64                                   100% |   9.0   B/s |   7.0   B |  00m01s
Complete!

```

**Comment by giuseppe** [comment]
that is fixed in crun 1.18.2

**Comment by benjamin-salchow** [comment]
Thank you @giuseppe that this if fixed in 1.18.2
